pkg/devfile/parser/data/1.0.0/components.go

```diff
@@ -6,35 +6,216 @@ import (
 	"github.com/openshift/odo/pkg/devfile/parser/data/common"
 )
 
-// GetComponents returns the slice of DevfileComponent objects parsed from the Devfile
+func (d *Devfile100) GetMetadata() common.DevfileMetadata {
+	// No GenerateName field in V2
+	return common.DevfileMetadata{
+		Name: d.Metadata.Name,
+		//Version: No field in V1
+	}
+}
+
+/// GetComponents returns the slice of DevfileComponent objects parsed from the Devfile
 func (d *Devfile100) GetComponents() []common.DevfileComponent {
-	return d.Components
+	var comps []common.DevfileComponent
+	for _, v := range d.Components {
+		comps = append(comps, convertV1ComponentToCommon(v))
+	}
+	return comps
 }
 
 // GetAliasedComponents returns the slice of DevfileComponent objects that each have an alias
 func (d *Devfile100) GetAliasedComponents() []common.DevfileComponent {
+	// TODO(adi): we might not need this for V2 as name is a required field now.
+	var comps []common.DevfileComponent
```

Comment:
Again, see other comment. These sections are all about Version 1. So why are we making changes / breaking backwards compatibility here?

Reply:
@cdrage Aditi can explain in more detail, but I think she had to make changes here to use the new common structs for both v1 and v2; so that v1 doesn't break

Reply:
@cdrage all v1 integration tests are passing, so it is the first sign we are not breaking anything.
secondly it is just a refactoring comment to remove this method from interface when we remove V1, as all components are aliased(they have name) for v2. agreed though comment is not clear.

Reply:
Then we should rename the folder / all mentions of v1.0.0 to something else in this folder as to be honest, we are implementing v2 features where v1 is..